docker/Dockerfile.pdo-tp-ccf

```diff
@@ -60,7 +61,8 @@ ENV PDO_INSTALL_ROOT=/pdo/dev/
 ENV PDO_SOURCE_ROOT=/pdo/private-data-objects/
 ENV PDO_ENCLAVE_CODE_SIGN_PEM=/pdo/private-data-objects/ccf_transaction_processor/enclave.pem
 ENV PDO_LEDGER_KEY_ROOT=/pdo/dev/opt/pdo/ccf/keys/
-ENV SGX_MODE=SIM
+ARG SGX_MODE=SIM
+ENV SGX_MODE=${SGX_MODE}
 ENV PDO_HOME=/pdo/dev/opt/pdo/
```

Comment:
Except for triggering the registration step in sim mode, are these necessary?

Reply:
thank you, in here for exactly the reason that you observed.

Reply:
Suggest we move to a canonical way of doing this. I've been adding a REBUILD arg that can be set to a timestamp in the makefile to force rebuild when PDO source changes. Seems a lot cleaner to be explicit. No changes requested at this time, but be careful about doing... unusual things just to make something obvious work.

Reply:
Moved to #415